metaserver/src/main/scala/scala/meta/languageserver/ScalametaLanguageServer.scala

```diff
@@ -242,9 +244,37 @@ class ScalametaLanguageServer(
       position: Position
   ): DefinitionResult = {
     val path = Uri.toPath(td.uri).get
-    symbolIndexer
-      .goToDefinition(path, position.line, position.character)
-      .getOrElse(DefinitionResult(Nil))
+    val locations = for {
+      symbol <- symbolIndexer.findSymbol(
+        path,
+        position.line,
+        position.character
+      )
+      data <- symbolIndexer.definitionData(symbol)
+      location <- symbolIndexer.definitionLocation(data)
+    } yield symbolIndexer.nonJarLocation(location)
+    DefinitionResult(locations.toList)
+  }
+
+  override def referencesRequest(
+      td: TextDocumentIdentifier,
+      position: Position,
+      context: ReferenceContext
+  ): ReferencesResult = {
+    val path = Uri.toPath(td.uri).get
+    val locations = for {
+      symbol <- symbolIndexer
+        .findSymbol(path, position.line, position.character)
+        .toList
+      data <- symbolIndexer.referencesData(symbol)
+      location <- symbolIndexer.referencesLocations(
+        data,
+        context.includeDeclaration
+      )
+    } yield location
+    // TODO(alexey) should it return references to the jars where symbol is defined? or should such references be filtered out instead?
```

Comment:
There's a different problem here: If you have `List` mentioned in your code and call find-references, you will get just references in _your code_. Then if you call go-to-definition, it will create a file in `target/sources/`. Now

Reply:
I think we have two options here

- create source files in target/source/* in references
- ignore references in jar:file uris

I lean towards the second option.

Reply:
Me too

Reply:
Can we update the comment to reflect that we intentionally skip jar:file uris?

Reply:
Done in 36123b8bd71107152b187e2f607e2006afd7a6b9